### ISSUE TITLE
[LLaDA-8B] Full forward and first block test fails (NaN logits) likely due to missing attention mask

### ISSUE THREAD

**Issue by jenriver** [jax-ml/bonsai#65]
The `test_full_forward_equivalence` and `first_block_equivalence` in TestLLaDAForwardPasses is currently failing, resulting in NaN arrays for the logits in the JAX model.

This failure is likely due to the lack of an attention mask being correctly applied during the forward pass, which is necessary to handle the padded sequences.

Currently marked as `@unittest.skip` to prevent CI failure, but needs revisiting.
